internal/consensus/state.go

```diff
@@ -1319,11 +1313,47 @@ func (cs *State) defaultDoPrevote(height int64, round int32) {
 		return
 	}
 
-	// Prevote cs.ProposalBlock
-	// NOTE: the proposal signature is validated when it is received,
-	// and the proposal block parts are validated as they are received (against the merkle hash in the proposal)
-	logger.Debug("prevote step: ProposalBlock is valid")
-	cs.signAddVote(tmproto.PrevoteType, cs.ProposalBlock.Hash(), cs.ProposalBlockParts.Header())
+	/*
+		22: upon <PROPOSAL, hp, roundp, v, −1> from proposer(hp, roundp) while stepp = propose do
+		23: if valid(v) && (lockedRoundp = −1 || lockedValuep = v) then
+		24: broadcast <PREVOTE, hp, roundp, id(v)>
+	*/
+	if cs.Proposal.POLRound == -1 {
```

Comment:
Any chance there can be a comment stating exactly what `POLRound == -1` implies in terms of the arxiv notation? Sorry if a dumb question 😅, Presumably its the `-1` in `<PROPOSAL, hp, roundp, v, −1>`, but its not obvious to me what that argument is when just reading this code in isolation.